ui/core.js

```diff
@@ -13,6 +13,8 @@
 //>>description: The core of jQuery UI, required for all interactions and widgets.
 //>>docs: http://api.jqueryui.com/category/ui-core/
 //>>demos: http://jqueryui.com/
+//>>css.structure: ../themes/base/core.css
```

Comment:
> The widgets will list its dependency on core.css themselves, e.g., dialog.js: //>>css.structure: base/core.css, base/dialog.css.

For option one yes thats correct.

> On regard of having CSS files to list their CSS dependencies, we could definitely do that. But, I would postpone it (as a new feature / next step), since this is not done on mobile either and it seems as an improvement rather than a requirement.

Agreed, this would be an enhancement. I was only saying its one potential solution to handle this.